test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -4021,5 +4021,236 @@ public void PackCommand_Deterministic_MultiplePackInvocations_CreateIdenticalPac
+        public void PackCommand_PackageIcon_HappyPath_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\icon.jpg", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("content\\icon.jpg")
+                .WithItem("None", "icon.jpg", null);
+
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_ImplicitFile_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\media\\icon.jpg", 10)
+                .WithFile("test\\media\\example.txt", 10)
+                .WithFile("test\\media\\readme.txt", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("media\\icon.jpg")
+                .WithItem("None", "media\\*", "media");
+
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_FolderNested_Suceeds()
+        {
+            var testDirBuilder = TestDirectoryBuilder.Create();
+            var projectBuilder = ProjectFileBuilder.Create();
+
+            testDirBuilder
+                .WithFile("test\\folder\\nested\\icon.jpg", 10)
+                .WithFile("test\\folder\\nested\\sample.txt", 10)
+                .WithFile("test\\folder\\nested\\media\\readme.txt", 10);
+
+            projectBuilder
+                .WithProjectName("test")
+                .WithPackageIcon("media\\nested\\icon.jpg")
+                .WithItem("None", "folder\\**", "media");
+                
+            using (var srcDir = testDirBuilder.Build())
+            {
+                projectBuilder.Build(msbuildFixture, srcDir.Path);
+                var result = msbuildFixture.PackProject(projectBuilder.ProjectFolder, projectBuilder.ProjectName, string.Empty);
+
+                ValidatePackIcon(projectBuilder);
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void PackCommand_PackageIcon_FolderNestedFlatPack_Suceeds()
```

Comment:
Similar to the above, I'm not sure what the specific scenario tested is.

Reply:
Similar to PackCommand_PackageIcon_FolderNested_Succeeds but, it tests that `PackagePath` is honored (puts the icon in the specified path/folder)

Reply:
Ok, I will de-dup with PackCommand_PackageIcon_FolderNested...

Reply:
I will only leave one happy path test, with additional asserts for the file/folder layouts.